packages/one-app-bundler/webpack/loaders/common.js

```diff
@@ -39,30 +39,72 @@ const cssLoader = ({ name = '', importLoaders = 2 } = {}) => ({
 
 /* eslint-disable inclusive-language/use-inclusive-words --
 config options for a third party library */
+// transform strings deep and greedy parameters into regex
+const reconcileSafeList = (safelist) => ({
+  ...safelist,
+  greedy: safelist.greedy
+    ? safelist.greedy.map((pattern) => new RegExp(pattern, 'i'))
+    : [],
+  deep: safelist.deep
+    ? safelist.deep.map((pattern) => new RegExp(pattern, 'i'))
+    : [/:global$/],
+});
+const reconcileWhiteList = (purgecss) => {
+  const newWhiteList = { aggregatedStandard: [], safelistDeep: [/:global$/] };
+  if (purgecss.whitelistPatterns) {
+    console.warn('Purgecss: Using depreciated property whitelistPatterns');
+    newWhiteList.aggregatedStandard = [
+      ...newWhiteList.aggregatedStandard,
+      ...purgecss.whitelistPatterns.map((pattern) => new RegExp(pattern, 'i')),
+    ];
+  }
+  if (purgecss.whitelist) {
+    console.warn('Purgecss: Using depreciated property whitelist');
+    newWhiteList.aggregatedStandard = [...newWhiteList.aggregatedStandard, ...purgecss.whitelist];
+  }
+  if (purgecss.whitelistPatternsChildren) {
+    console.warn('Purgecss: Using depreciated property whitelistPatternsChildren');
+
+    newWhiteList.safelistDeep = purgecss.whitelistPatternsChildren.map((pattern) => new RegExp(pattern, 'i'));
+  }
+  return newWhiteList;
+};
+
 const purgeCssLoader = () => {
-  const configOptions = getConfigOptions();
-  const whitelistPatterns = configOptions.purgecss.whitelistPatterns
-    ? configOptions.purgecss.whitelistPatterns.map((pattern) => new RegExp(pattern, 'i'))
-    : [];
-  const whitelistPatternsChildren = configOptions.purgecss.whitelistPatternsChildren
-    ? configOptions.purgecss.whitelistPatternsChildren.map((pattern) => new RegExp(pattern, 'i'))
-    : [/:global$/];
-  if (configOptions.purgecss.disabled) return [];
+  const { purgecss } = getConfigOptions();
+  if (purgecss.disabled) return [];
+
+  let aggregatedWhiteList = { aggregatedStandard: [], safelistDeep: [/:global$/] };
```

Comment:
```suggestion
  let aggregatedAllowList = { aggregatedStandard: [], safelistDeep: [/:global$/] };
```
where possible replace whiteList with allowList